src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,223 @@
+    IConvexBooster public constant BOOSTER = IConvexBooster(0xF403C135812408BFbE8713b5A23a04b3D48AAE31);
+
+    // Pools
+    uint256 public poolLength;
+
+    mapping(address => PoolInfo) public pools;
+
+    // Deployed RepresentingConvexTokens, mapping(Curve LP token => representing Convex token)
+    mapping(address => address) public deployedTokens;
+
+    // Errors
+    error InvalidAssetType();
+    error UnknownAssetA();
+
+    constructor(address _rollupProcessor) BridgeBase(_rollupProcessor) {
+        loadPools();
+
+        uint256[] memory criterias = new uint256[](poolLength);
+        uint32[] memory gasUsage = new uint32[](poolLength);
+        uint32[] memory minGasPerMinute = new uint32[](poolLength);
+
+        for (uint256 i = 0; i < poolLength; i++) {
+            (address curveLpToken, , , , , ) = BOOSTER.poolInfo(i);
+            criterias[i] = uint256(keccak256(abi.encodePacked(curveLpToken, deployedTokens[curveLpToken])));
+            gasUsage[i] = 500000; // deposit 1M, withdrawal 375k -> compromise 500k
+            minGasPerMinute[i] = 350;
+        }
+
+        SUBSIDY.setGasUsageAndMinGasPerMinute(criterias, gasUsage, minGasPerMinute);
+    }
+
+    /**
+    @notice function so the bridge can receive ether. Used for subsidy.
+    */
+    receive() external payable {}
+
+    /**
+    @notice Stake and unstake Curve LP tokens through Convex Finance Booster anytime.
+    @notice Convert rate between Curve LP token and corresponding Convex LP token is 1:1
+    @notice Stake == Deposit, Unstake == Withdraw
+    @param _inputAssetA Curve LP token (staking), virtual asset (unstaking)
+    @param _outputAssetA Virtual asset (staking), Curve LP token (unstaking)
+    @param _totalInputValue Total number of Curve LP tokens to deposit / withdraw
+    @param outputValueA Number of Curve LP tokens staked / unstaked
+    @param _rollupBeneficiary Address of the contract that receives subsidy
+    */
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata _inputAssetB,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata _outputAssetB,
+        uint256 _totalInputValue,
+        uint256,
+        uint64,
+        address _rollupBeneficiary
+    )
+        external
+        payable
+        override(BridgeBase)
+        onlyRollup
+        returns (
+            uint256 outputValueA,
+            uint256,
+            bool
+        )
+    {
+        if (_totalInputValue == 0) {
+            revert ErrorLib.InvalidInputAmount();
+        }
+
+        if (
+            _inputAssetA.assetType != AztecTypes.AztecAssetType.ERC20 ||
+            _outputAssetA.assetType != AztecTypes.AztecAssetType.ERC20
+        ) {
+            revert InvalidAssetType();
+        }
+
+        if (
+            deployedTokens[_inputAssetA.erc20Address] == address(0) &&
+            deployedTokens[_outputAssetA.erc20Address] == address(0)
```

Comment:
You are accessing storage variables on the 2 lines above and then again on line 131 and 143. Accessing storage is expensive so pls pre-load it to local variables and then do the checks on those.

Reply:
Resolved